backend/tournesol/tests/test_api_criteria_correlation.py

```diff
@@ -0,0 +1,110 @@
+from rest_framework import status
+from rest_framework.test import APIClient
+
+from core.tests.factories.user import UserFactory
+from tournesol.models.ratings import ContributorRatingCriteriaScore
+from tournesol.tests.factories.entity import EntityFactory
+from tournesol.tests.factories.poll import (
+    CriteriaFactory,
+    CriteriaRankFactory,
+    PollFactory,
+    PollWithCriteriasFactory,
+)
+from tournesol.tests.factories.ratings import (
+    ContributorRatingCriteriaScoreFactory,
+    ContributorRatingFactory,
+)
+from tournesol.views.criteria_correlations import compute_correlation, compute_slope
+
+
+def test_compute_correlation():
+    assert compute_correlation({}, {}) == None
+    assert compute_correlation({"a": 1}, {"b": 1}) == None
+    assert compute_correlation({"a": 1}, {"a": 1}) == None
+    assert np.allclose(compute_correlation({"a": 1, "b": 2}, {"a": 1, "b": 2}), 1)
+    assert np.allclose(compute_correlation({"a": 1, "b": -2}, {"a": -1, "b": 2}) ,-1)
+    assert np.allclose(compute_correlation({"a": 1, "b": -2, "c": -2, "d": 0}, {"a": -1, "b": 2, "c": 2, "d": 0}) ,-1)
+    assert compute_correlation({"a": 1, "b": 2, "c": 0}, {"a": 3, "b": 3, "c": 1}) < 1
+    # correlation is not defined in the below case 
+    # because all scores on one of the criteria are equal
+    assert compute_correlation({"a": 1, "b": 2}, {"a": 3, "b": 3}) == None
+
+
+def test_compute_slope():
+    assert compute_slope({}, {}) == None
+    assert compute_slope({"a": 1}, {"b": 1}) == None
+    assert compute_slope({"a": 1}, {"a": 1}) == None
+    assert compute_slope({"a": 1, "b": -2}, {"a": -1, "b": 2}) == -1
+    assert compute_slope({"a": 1, "b": 2}, {"a": 3, "b": 3}) == 0
+
+    entities="qwertzuiopasdfghjklyxcvbnm"
+    assert np.allclose(compute_slope(
+        {e: 42 - 2 * i for i, e in enumerate(entities)}, 
+        {e: 24 + 3 * i for i, e in enumerate(entities)}, 
+    ), -3/2)
+    assert np.allclose(compute_slope(
+        {e: i for i, e in enumerate(entities)}, 
+        {e: 24 + 3 * i for i, e in enumerate(entities)}, 
+    ), 3)
+
+
+
+class CorrelationAPI(TestCase):
+
+    def setUp(self):
+        self.user = UserFactory(username="user_test_correlation")
+        self.client = APIClient()
+        self.client.force_authenticate(user=self.user)
+
+        self.poll = PollFactory()
+        criterias = [
+            CriteriaRankFactory(criteria__name="hello", poll=self.poll, rank=1),
+            CriteriaRankFactory(criteria__name="world", poll=self.poll, rank=2),
+            CriteriaRankFactory(criteria__name="!", poll=self.poll, rank=3)
+        ]
+        self.entities = EntityFactory.create_batch(10)
+        self.ratings = [ContributorRatingFactory(user=self.user, entity=entity) for entity in self.entities]
+
+    def test_correlations_no_data(self):
+        response = self.client.get(f"/users/me/criteria_correlations/{self.poll.name}/")
+        assert response.status_code == status.HTTP_200_OK
```

Comment:
Most of our tests use `self.assertEqual` from Django.

This new test case "breaks" the coherence with the rest of the application, and contributors may be confused.

If this test case is not a proposition to switch from `self.assertEqual` to `assert`, I propose to stick to the current style for now (see the test_api_comparison.py file for instance).

Reply:
Done in e2ba66d3